.github/autobuild/linux_deb.sh

```diff
@@ -61,18 +64,22 @@ setup() {
 }
 
 setup_cross_compilation_apt_sources() {
-    if [[ "${TARGET_ARCH}" == amd64 ]]; then
+    if [[ "${TARGET_ARCH}" == "${APT_ARCH}" ]]; then
         return
     fi
+    echo "Building on ${APT_ARCH} for ${TARGET_ARCH}. Thus modifying apt..."
     sudo dpkg --add-architecture "${TARGET_ARCH}"
-    # Duplicate the original Ubuntu sources and modify them to refer to the TARGET_ARCH:
-    sed -rne "s|^deb.*/ ([^ -]+(-updates)?) main.*|deb [arch=${TARGET_ARCH}] http://ports.ubuntu.com/ubuntu-ports \1 main universe multiverse restricted|p" /etc/apt/sources.list | sudo dd of=/etc/apt/sources.list.d/"${TARGET_ARCH}".list
-    # Now take the original Ubuntu sources and limit those to the build host (i.e. non-TARGET_ARCH) architectures:
-    sudo sed -re 's/^deb /deb [arch=amd64,i386] /' -i /etc/apt/sources.list
+
+    if [[ "${APT_ARCH}" == "amd64"  ]]; then
```

Comment:
What happens if `"${APT_ARCH}"` is `arm64` and `"${TARGET_ARCH}"` is `"amd64"`, for example?  Should this _explicitly error_ before this point and then take `"${APT_ARCH}" == "amd64"` as given?

Reply:
That's a very good question. The present logic here seems to assume that cross-compilation is only done on an `amd64` runner. That's no longer the case: we explicitly cross-compile from `arm64` to `armhf` on an ARM runner. So I wonder whether this arch duplication is still needed in that case?

It would be good to get a reference for where this technique came from and why/when it's needed. Is @hoffie still around?